src/react/components/StyleForm.stories.js

```diff
@@ -0,0 +1,158 @@
+        }
+    },
+    "stroke": { // stroke styles
+        "color": "#000000", // stroke color
+        "width": 3, // stroke width
+        "lineDash": "solid", // line dash, supported: dash, dashdot, dot, longdash, longdashdot and solid
+        "lineCap": "round", // line cap, supported: mitre, round and square
+        "area": {
+            "color": "#000000", // area stroke color
+            "width": 3, // area stroke width
+            "lineDash": "solid", // area line dash
+            "lineJoin": "round" // area line corner
+        }
+    },
+    "text": { // text style
+        "fill": { // text fill style
+            "color": "#000000" // fill color
+        },
+        "stroke": { // text stroke style
+            "color": "#ffffff", // stroke color
+            "width": 1 // stroke width
+        },
+        "font": "bold 12px Arial", // font
+        "textAlign": "top", // text align
+        "offsetX": 12, // text offset x
+        "offsetY": 12, // text offset y
+        "labelText": "example", // label text
+        "labelProperty": "propertyName" // read label from feature property
+    },
+    "image": { // image style
+        "shape": 5, // 0-6 for default markers. svg or external icon path
+        "size": 3, // Oskari icon size.
+        "sizePx": 20, // Exact icon px size. Used if 'size' not defined.
+        "offsetX": 0, // image offset x
+        "offsetY": 0, // image offset y
+        "opacity": 0.7, // image opacity
+        "radius": 2, // image radius
+        "fill": {
+            "color": "#ff00ff" // image fill color
+        }
+    },
+    "inherit": false, // For hover. Set true if you wan't to extend original feature style.
+    "effect": "auto normal" // Requires inherit: true. Lightens or darkens original fill color. Values [darken, lighten, auto] and [minor, normal, major].
+};
+
+const testOptions = [
+    {
+        value: 'testi tyyli dd',
+        format: 'area',
+    },
+    {
+        value: 'tyyli 2',
+        format: 'line'
+    },
+    {
+        value: 'storybooktyyli',
+        format: 'area'
+    }
+];
+
+const styleList = [
+    {
+        value: 'testi tyyli ff',
+        format: 'point',
+        fill: {
+            color: '#c5c5c5',
+            area: {
+                pattern: 'solid'
+            }
+        },
+        stroke: {
+            color: "#008720",
+            width: 3,
+            lineDash: "solid",
+            lineCap: "round",
+            area: {
+                color: "#008720",
+                width: 3,
+                lineDash: "solid",
+                lineJoin: "round"
+            }
+        },
+        image: {
+            shape: 2,
+            size: 4,
+        }
+    },
+    {
+        value: 'tyyli 2',
+        format: 'line',
+    },
+    {
+        value: 'storybooktyyli',
+        format: 'area',
+    }
+];
+
+
+storiesOf('StyleForm', module)
+    .add('Default', () => (
+        <StyleForm
+            icons={ testOptions }
```

Comment:
Not sure what these icons do or why they need to be passed. I think we only need to pass:
- style json
- markers (so the user can select which point symbol to use for points)
- a controller that will have setters for the individual controls on the form to call OR an onChange callback that will get the updated style when the user changes anything on the form (that should trigger a new render on the storybook that uses that updated style as prop, initializing a new version of the form)

Reply:
Fixed.